plugins/http_plugin/tests/unit_tests.cpp

```diff
@@ -623,7 +623,7 @@ BOOST_FIXTURE_TEST_CASE(bytes_in_flight, http_plugin_test_fixture) {
    };
 
    auto wait_for_no_bytes_in_flight = [&](uint16_t max = std::numeric_limits<uint16_t>::max()) {
-      while (http_plugin->bytes_in_flight() > 0 && http_plugin->requests_in_flight() > 0 && --max)
```

Comment:
Seems like this shouldn't make a difference as they should both either be 0 or both non-zero.

Reply:
I'm confused by that too. And I don't understand the discrepancy why `wait_for_no_bytes_in_flight` looks at both while `wait_for_no_requests_in_flight` looks at only one. If the intent is to look at both then a common `wait_for_nothing_in_flight` could be made that looks at both.

Reply:
In practice the requests in flight also matter. The issue is that bytes in flight can reach 0 before all the requests are done (or started).

`bytes_in_flight` is only incremented on sending the response. We do not, currently at least, include the request.
https://github.com/AntelopeIO/spring/blob/2b43faf02056ed303768708ea9e4152cadc1a56b/plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp#L466-L468

Therefore, requests can be in progress, but `bytes_in_flight` can be 0. So we need to wait until all requests are complete to know that all `bytes_in_flight` are accounted for.

Reply:
So could we just wait for `requests_in_flight()` to be 0, and then assert that `bytes_in_flight()` is 0 as well?

Reply:
Sure, that sounds better